discovery/src/main/java/io/airlift/discovery/client/DiscoveryBinder.java

```diff
@@ -104,15 +117,90 @@ public void bindHttpSelector(ServiceType serviceType)
         binder.bind(HttpServiceSelector.class).annotatedWith(serviceType).toProvider(new HttpServiceSelectorProvider(serviceType.value())).in(Scopes.SINGLETON);
     }
 
+    public static class HttpServiceAnnouncementBuilder
+    {
+        private final MapBinder<String, String> propertiesBinder;
+
+        public HttpServiceAnnouncementBuilder(MapBinder<String, String> propertiesBinder)
+        {
+            this.propertiesBinder = propertiesBinder;
+        }
+
+        public PropertyBinding bindProperty(String key)
+        {
+            requireNonNull(key, "key is null");
+            return new PropertyBinding()
+            {
+                @Override
+                public HttpServiceAnnouncementBuilder toProvider(Provider<String> provider)
+                {
+                    propertiesBinder.addBinding(key).toProvider(provider);
+                    return HttpServiceAnnouncementBuilder.this;
+                }
+
+                @Override
+                public HttpServiceAnnouncementBuilder toProvider(Key<? extends Provider<String>> providerKey)
+                {
+                    propertiesBinder.addBinding(key).toProvider(providerKey);
+                    return HttpServiceAnnouncementBuilder.this;
+                }
+            };
+        }
+
+        @CanIgnoreReturnValue
+        public HttpServiceAnnouncementBuilder addProperty(String key, String value)
+        {
+            bindProperty(key).toInstance(value);
+            return this;
+        }
+
+        @CanIgnoreReturnValue
+        public HttpServiceAnnouncementBuilder addProperties(Map<String, String> properties)
+        {
+            properties.forEach(this::addProperty);
+            return this;
+        }
+
+        public interface PropertyBinding
+        {
+            @CanIgnoreReturnValue
+            default HttpServiceAnnouncementBuilder toInstance(String value)
```

Comment:
yes, it's redundant, but more consistent with how binders generally work

i think it could be unfriendly to type `.bindProperty("foo") ...` and see code completion for more sophisticated use, but not for the less sophisticated.

we can deprecate `addProperty` if needed.

or remove `bindProperty` replacing it with `addProperty` overloads that take a provider